pydicom/dataset.py

```diff
@@ -1754,6 +1776,351 @@ def walk(self, callback, recursive=True):
                     for dataset in sequence:
                         dataset.walk(callback)
 
+    @classmethod
+    def _convert_to_python_number(cls, value, vr):
+        """Makes sure that values are either ints or floats
+        based on their value representation.
+
+        Parameters
+        ----------
+        value: Union[Union[str, int, float], List[Union[str, int, float]]]
+            value of data element
+        vr: str
+            value representation of data element
+
+        Returns
+        -------
+        Union[Union[str, int, float], List[Union[str, int, float]]]
+
+        """
+        if value is None:
+            return None
+        number_type = None
+        if vr in cls._VRs_TO_BE_INTS:
+            number_type = int
+        if vr in cls._VRs_TO_BE_FLOATS:
+            number_type = float
+        if number_type is not None:
+            if isinstance(value, (list, tuple, )):
+                value = [number_type(e) for e in value]
+            else:
+                value = number_type(value)
+        return value
+
+    @classmethod
+    def _data_element_from_json(cls, tag, vr, value, value_key,
+                                bulk_data_uri_handler=None):
+        """Creates a DataElement from JSON.
```

Comment:
Is it possible to have this be a class method of DataElement?  Would seem like a more natural fit, and might allow for other people to build their own custom code.  It does use cls() (i.e. Dataset or derivative), but perhaps that could be passed in?